packages/studio/src/components/AddPageButton/EntityIdField.tsx

```diff
@@ -0,0 +1,125 @@
+import Select, { MultiValue } from "react-select";
+import useStudioStore from "../../store/useStudioStore";
+import StreamScopeFieldLabel from "./StreamScopeFieldLabel";
+import { ChangeEvent, useCallback, useMemo, useState } from "react";
+import classNames from "classnames";
+import getSelectCssClasses from "../../utils/getSelectCssClasses";
+
+interface Props {
+  disabled: boolean;
+  updateSelection: (selectedIds: string[]) => void;
+  selectedIds?: string[];
+}
+
+export default function EntityIdField({
+  disabled,
+  updateSelection,
+  selectedIds,
+}: Props) {
+  const [entitiesRecord, fetchEntities] = useStudioStore((store) => [
+    store.accountContent.entitiesRecord,
+    store.accountContent.fetchEntities,
+  ]);
+  const availableEntityTypes = Object.entries(entitiesRecord)
+    .filter(([_entityType, entityData]) => {
+      return Object.values(entityData).length > 0;
+    })
+    .map((entry) => entry[0]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [entityType, setEntityType] = useState<string | undefined>(
+    availableEntityTypes[0]
+  );
+  const entityOptions = useEntityOptions(entityType);
+  const selectedOptions = useMemo(
+    () => selectedIds?.map((id) => ({ value: id, label: id })),
```

Comment:
I also think it's a little more transparent this way because this is the information you end up writing in your TemplateConfig. If product asks for it I'd make it a separate item

Reply:
I think I'd want this to be consistent with the saved filters, so we'd either remove the display name from selected saved filters to match, or add the display name here. I agree that having IDs appear in the input make sense because that's what they're really choosing for the template config, but it also makes it harder to compare against the dropdown when adding multiple filters

Reply:
Would you be okay deferring this until we get to the corresponding edit page settings updates? I think we might have a better idea at that point how to proceed